libs/langchain/langchain/document_loaders/batch_loader.py

```diff
@@ -0,0 +1,252 @@
+    documents = loader_youtube.load(method="sequential")
+    documents = loader_youtube.load(method="async")
+    ```
+    """
+
+    def __init__(
+        self,
+        loader_initializer_callable: Callable[..., BaseLoader],
+        init_loader_args: Dict[str, List[Any]],
+        show_progress: bool = False,
+    ):
+        """Initialize the loader with a loader_callable (either a class or a
+        function used to initialize the loader) and a list of loader_args.
+
+        Args:
+            loader_initializer_callable (Callable[..., BaseLoader]): callable
+                used to initialize the loader (class or init function )
+            init_loader_args (Dict[str, List[Any]]): list of loader_args used
+                to initialize the loader. The keys of the dict are the name of
+                the arguments and the values are the list of values for each.
+            show_progress: If true, will show a progress bar as the files are
+                loaded. This forces an iteration through all matching files to
+                count them prior to loading them.
+
+        Raises:
+            ValueError: if the length of the lists in loader_args are not equal
+        """
+        loader_args = list(zip(*init_loader_args.values()))
+        self.loaders_args = [
+            dict(zip(init_loader_args.keys(), loader_arg)) for loader_arg in loader_args
+        ]
+        self.loader_callable = loader_initializer_callable
+        self.show_progress = show_progress
+        self._partial_load = partial(
+            self._load_loader, loader_callable=self.loader_callable
+        )
+
+        if len(set(len(arg) for arg in init_loader_args.values())) != 1:
+            raise ValueError("All loader_args lists must have the same length")
+
+    def load(self, method: str = "sequential", max_workers: int = 1) -> List[Document]:
+        """Load data from the loader using the specified method. The list of
+        documents is flattened.
+
+        Possible methods are:
+            - "sequential": load data sequentially
+            - "thread": load data using threads
+            - "process": load data using processes
+            - "async": load data using asyncio
+
+        Args:
+            method (str, optional): concurrent method to use.
+                Defaults to "sequential".
+            max_workers (int, optional): number of workers if 'thread' or
+                'process' methods are used. Defaults to 1.
+
+        Raises:
+            ValueError: if the method is not valid
+
+        Returns:
+            List[Document]: list of loaded documents
+        """
+        if method == "thread":
+            return self._load_thread(max_workers)
+        elif method == "process":
+            return self._load_process(max_workers)
+        elif method == "sequential":
+            return self._load_sequential()
+        elif method == "async":
+            return asyncio.run(self._async_load())
```

Comment:
Could we kick off the async code from a thread if there's an existing event loop so users don't have to use nest_asyncio?

Reply:
You're right. Do you think the best solution would be to have another async method ?
E.g

async def aload()
[...]
def load()
[...]